### PR TITLE
tests-for-shopping-cart

### PR DESCRIPTION
**Pull request by margosalii**

### CODE REVIEW

File: src/main/java/mate/academy/store/model/ShoppingCart.java [hunk 1]
```diff
@@ -23,6 +24,7 @@
 @SQLRestriction("is_deleted=false")
 @Getter
 @Setter
+@EqualsAndHashCode
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/store/model/ShoppingCart.java:27]: If you add these annotation, you need to exclude cartItems
@EqualsAndHashCode.Exclude

File: src/test/java/mate/academy/store/controller/ShoppingCartControllerTest.java [hunk 2]
```diff
@@ -0,0 +1,277 @@
...
+        MvcResult result = mockMvc.perform(
+                get("/cart")
+                    .contentType(MediaType.APPLICATION_JSON))
+                .andExpect(status().isOk())
+                .andReturn();
+
+        ShoppingCartDto actual = objectMapper
+                .readValue(result.getResponse().getContentAsString(), ShoppingCartDto.class);
+
+        Assertions.assertEquals(expected, actual);
```
Comment (Elena-Bruyako) [inline src/test/java/mate/academy/store/controller/ShoppingCartControllerTest.java]: use static imports

File: src/test/java/mate/academy/store/controller/ShoppingCartControllerTest.java [hunk 3]
```diff
@@ -0,0 +1,277 @@
...
+                "classpath:database/books/delete-books.sql"
+            },
+            executionPhase = Sql.ExecutionPhase.AFTER_TEST_METHOD
+    )
+    void deleteCartItemById_Ok() throws Exception {
+        mockMvc.perform(
+                delete("/cart/items/1"))
+                .andExpect(status().isOk())
+                .andReturn();
+    }
```
Comment (Elena-Bruyako) [inline src/test/java/mate/academy/store/controller/ShoppingCartControllerTest.java:215]: check if you need not valid test cases (not valid id for example)

**Review by Elena-Bruyako — CHANGES_REQUESTED** [review]
good job!

File: src/test/java/mate/academy/store/controller/ShoppingCartControllerTest.java (hunk 3 above)
Comment (margosalii) [reply src/test/java/mate/academy/store/controller/ShoppingCartControllerTest.java:215]: > check if you need not valid test cases (not valid id for example)

i tried something like that, but i don't know how to fix that problem
image

File: src/test/java/mate/academy/store/controller/ShoppingCartControllerTest.java [hunk 4]
```diff
@@ -0,0 +1,276 @@
...
+    @Sql(
+            scripts = {
+                "classpath:database/cart.items/delete-cart-item.sql",
+                "classpath:database/shopping.carts/delete-shopping-cart.sql",
+                "classpath:database/categories/delete-categories-and-book-categories.sql",
+                "classpath:database/categories/delete-categories.sql",
+                "classpath:database/books/delete-books.sql"
+            },
+            executionPhase = Sql.ExecutionPhase.AFTER_TEST_METHOD
+    )
```
Comment (okuzan) [inline src/test/java/mate/academy/store/controller/ShoppingCartControllerTest.java]: extract scripts common to each test of class to the class level

File: src/test/resources/database/cart.items/add-cart-item.sql [hunk 5]
```diff
@@ -0,0 +1 @@
+insert into cart_items (id, shopping_cart_id, book_id, quantity) values (1, 1, 1, 2);
```
Comment (okuzan) [inline src/test/resources/database/cart.items/add-cart-item.sql]: image
minor issue but when you see red signs, add an empty line at EOF

File: src/test/java/mate/academy/store/service/shopping/cart/ShoppingCartServiceTest.java [hunk 6]
```diff
@@ -0,0 +1,216 @@
...
+        ShoppingCartDto actual = shoppingCartService.getShoppingCart(ID);
+
+        assertEquals(expected, actual);
+    }
+
+    @Test
+    void getShoppingCart_byInvalidId_ExceptionOk() {
+        when(shoppingCartRepository.findByUserId(anyLong())).thenReturn(Optional.empty());
+        assertThrows(EntityNotFoundException.class, () -> shoppingCartService.getShoppingCart(ID));
+
```
Comment (okuzan) [inline src/test/java/mate/academy/store/service/shopping/cart/ShoppingCartServiceTest.java]:
```suggestion
```

**Review by okuzan — APPROVED** [review]

**Pull request merged**